### PR TITLE
Update readme.md

### PR DESCRIPTION
**Pull request by sesam**
Tried to reorganise to make it easier to digest. Good?

### CODE REVIEW

File: readme.md [hunk 1]
```diff
@@ -3,14 +3,26 @@ LÖVE is an *awesome* framework you can use to make 2D games in Lua. It's free,
...
-If you need further help, feel free to ask on our [forums][forums], our [Discord server][discord], or our IRC channel [#love on OFTC][irc].
+Come join us on our [forums][forums], our [Discord chat][discord], or our IRC channel [#love on OFTC][irc] and feel free to ask us anything.
 
-Compilation
------------
+Contributing
+------------
+
+Please contribute by looking through the [open issues][issues] and the forums and chat.
+Ready to make a code contribution? Very welcome! Please also read the [source code style guide][codestyle].
```
Comment (zorggn) [inline readme.md]: "Very welcome!" as a phrase sounds a bit weird to me... either something like "You're welcome to do so!" or "We'd be glad!" or something...

**Comment by MikuAuahDark** [comment]
IMO the current readme is as good as it gets. I'll let slime to decide.

File: readme.md (hunk 1 above)
Comment (sesam) [reply readme.md]: Agreed. I think we can find better phrasing, to make it feel like newconers are very welcomed. The original text works as long as PRs are really welcome and users don't get too hindered by the code style requirements. Automatic code style checks is a good way to solve that, and maybe style checks are already automated?

**Comment by sesam** [comment]
I think this newly copied issue, unless closed :), serves as a placeholder for the opinion that the readme might still be less than complete and unchangeable :).
On the other hand @mikuAuahDark if you have read this PR, do you have something specific that you feel is bad, wrong, or that you wish that it does not get changed in the near future?

**Comment by slime73** [comment]
> I think this newly copied (?) issue from
https://bitbucket.org/rude/love/issues/1519/installation-help-on-frontpage serves as a placeholder for the opinion that the readme might still not be final

While the readme isn't set in stone, that issue was about the love2d.org homepage rather than the repository homepage.

File: readme.md [hunk 2]
```diff
@@ -3,14 +3,26 @@ LÖVE is an *awesome* framework you can use to make 2D games in Lua. It's free,
 [![Build Status: Windows](https://ci.appveyor.com/api/projects/status/chc0hdr08wv1d5c7?svg=true)](https://ci.appveyor.com/project/AlexSzpakowski/love)
 [![Build Status: Github CI](https://github.com/love2d/love/workflows/continuous-integration/badge.svg)](https://github.com/love2d/love/actions?query=workflow%3Acontinuous-integration)
 
-Documentation
--------------
+Please see our [living wiki documentationwiki][wiki] for deeper explanations and documentation of this framework.
```
Comment (slime73) [inline readme.md]: I don't think it's necessary to remove the Documentation section, there's also a typo here and "living" is redundant when talking about a wiki. I think the Documentation section was fine as-is.

File: readme.md [hunk 3]
```diff
@@ -3,14 +3,26 @@ LÖVE is an *awesome* framework you can use to make 2D games in Lua. It's free,
...
-We use our [wiki][wiki] for documentation.
-If you need further help, feel free to ask on our [forums][forums], our [Discord server][discord], or our IRC channel [#love on OFTC][irc].
+Come join us on our [forums][forums], our [Discord chat][discord], or our IRC channel [#love on OFTC][irc] and feel free to ask us anything.
 
-Compilation
------------
+Contributing
+------------
+
+Please contribute by looking through the [open issues][issues] and the forums and chat.
```
Comment (slime73) [inline readme.md]: The phrasing here kind of misses the original intent of what was there before, I think. Posting an issue and talking about potential changes in the forums and chat channels are a form of contribution. Replying to an existing issue in the tracker is as well, and I think the original text covers all of those cases.

The original phrasing kind of makes it seem like pull requests aren't a part of the above "best places" to contribute so a change to that phrasing might be helpful. As zorg says, the new phrasing is weird though.

 Also I'm not sure whether it should be explicitly pointed out in the readme or not, but for any large-scale or API-related change, discussing it through the forums/chat or the issue tracker is usually considered good manners before any pull request is made.

File: readme.md [hunk 4]
```diff
@@ -3,14 +3,26 @@ LÖVE is an *awesome* framework you can use to make 2D games in Lua. It's free,
...
+Ready to make a code contribution? Very welcome! Please also read the [source code style guide][codestyle].
+
+## Git branches
```
Comment (slime73) [inline readme.md]: This section doesn't just cover branches so I think the original header is more accurate.

File: readme.md (hunk 1 above)
Comment (sesam) [reply readme.md]: I ended up not changing this wording.

File: readme.md (hunk 2 above)
Comment (sesam) [reply readme.md]: Right. I ended up canceling out this change so both headline and text stays unchanged.

File: readme.md (hunk 1 above)
Comment (zorggn) [reply readme.md]: Either grammatically or semantically, it's still wrong. What's exactly welcome? The user? The changes made? I'm sure the last thing we want is people to find issue with the non-englishness of the readme file. :3

File: readme.md (hunk 3 above)
Comment (sesam) [reply readme.md]: Right -- I've changed back so the first sentence is as it was, just with an added direct link to the open issues page.
The second sentence is now changed from

For code contributions, pull requests and patches are welcome. Be sure to read the [source code style guide][codestyle].

To:

Ready to make a code contribution? Excellent! Please come discuss any bigger changes in the chat, and do also check our [source code style guide][codestyle].

File: readme.md (hunk 4 above)
Comment (sesam) [reply readme.md]: 👍 changed title back, so no change here anymore.

**Comment by sesam** [comment]
Thank you @slime73  I probably was looking for something else originally. On the other hand I like to read "actively" in a text editor and try to improve whatever I read as I go.
With a second reading, and taking into account comments, I've reversed a couple of changes, and I hope all problems and improvements (non-improvements! :)) raised so far have been taken care of. 

The "graphical diff" for markdown gives quite a good overview:  https://github.com/love2d/love/pull/1550/files#diff-0730bb7c2e8f9ea2438b52e419dd86c9

Basically this:
1. the order is changed, so the reader gets info on contributing, then location of various versions of the code, then how to compile.
2. Adds a direct link to issues page
3. Rephrased a bit and increased readability., just few touches based on what I remember from school and from having longer ago experimented with tools similar to readable.com

Is this updated change-set an improvement on the existing readme? Is there something more that should or should not be changed? Improved?

File: readme.md (hunk 1 above)
Comment (sesam) [reply readme.md]: Of course, so right you are. I'm not a native English speaker and I'm bound to make mistakes. I also do value that many potential contributors are not native English speakers, so we would be wise to take a constructive stance towards making it easy for people from all walks of life to take part and contribute. The first step towards such a goal is to establish a welcoming atmosphere.
Please, do feel free to give constructive suggestions on how to further improve on the choice of words, grammar, structure, etc. and to your hearts content! Pray, do have a very nice day!

Comment (sesam) [reply readme.md]: The new sentence in question is, since some hours ago, changed to read like this:

> Ready to make a code contribution? Excellent! Please come discuss any bigger changes in the chat, and do also check our [source code style guide][codestyle].

Maybe there's an implicit "and" in there that I ought to make explicit? What's your take?

**Comment by slime73** [comment]
I made some of my own changes here https://github.com/love2d/love/commit/774ebfcf90db7d1a543116ff593849675ada7b5f, which supersedes this change.

If people have other tweaks or suggestions for the readme maybe they can be discussed on Discord or in an issue or followup PR.
